tests/stress_tests/memcheck_tests/tests_pipelines/tests_pipelines.cpp

```diff
@@ -14,127 +14,193 @@
-
-#define log_info_cur_mem_usage()                                                                                    \
-        log_info("Current values of virtual memory consumption:");                                                  \
-        log_info("VMRSS\t\tVMHWM\t\tVMSIZE\t\tVMPEAK");                                                                   \
-        log_info(test_cur_vmrss << "\t\t" << test_cur_vmhwm << "\t\t" << test_cur_vmsize << "\t\t" << test_cur_vmpeak);
+
+class MemCheckPipeline {
+private:
+    std::array<std::string, MeasureValueMax> measures_headers = {{"VMRSS", "VMHWM", "VMSIZE", "VMPEAK", "THREADS"}};
+    std::array<long, MeasureValueMax> measures;            // current measures
+    std::array<long, MeasureValueMax> start_measures;      // measures before run (will be used as baseline)
+public:
+    MemCheckPipeline() {
+        start_measures[VMRSS] = (long) getVmRSSInKB();
+        start_measures[VMHWM] = start_measures[VMRSS];
+        start_measures[VMSIZE] = (long) getVmSizeInKB();
+        start_measures[VMPEAK] = start_measures[VMSIZE];
+        start_measures[THREADS] = (long) getThreadsNum();
+
+        std::copy(start_measures.begin(), start_measures.end(), measures.begin());
+    }
+
+    ~MemCheckPipeline() {
+        // Backward compatible information. Required for DB data upload
+        log_info("[MEASURE] Current values of virtual memory consumption:");
+        log_info(get_measures_headers_as_str("\t\t"));
+        log_info(get_measures_as_str("\t\t"));
+    }
+
+    void do_measures() {
+        measures[VMRSS] = (long) getVmRSSInKB();
+        measures[VMHWM] = (long) getVmHWMInKB();
+        measures[VMSIZE] = (long) getVmSizeInKB();
+        measures[VMPEAK] = (long) getVmPeakInKB();
+        measures[THREADS] = (long) getThreadsNum();     // TODO: resolve *-32295
+    }
+
+    std::array<long, MeasureValueMax> get_measures() {
+        std::array<long, MeasureValueMax> temp;
+        std::transform(std::begin(measures), std::end(measures), std::begin(start_measures), std::begin(temp),
+                       [](long measure, long start_measure) -> long {
+                           return measure - start_measure;
+                       });
+
+        return temp;
+    }
+
+    std::string get_measures_as_str(std::string delimeter) {
+        std::array<long, MeasureValueMax> temp = get_measures();
+        std::string str = std::to_string(*temp.begin());
+        for (auto it = temp.begin() + 1; it != temp.end(); it++) {
+            str += delimeter + std::to_string(*it);
+        }
+        return str;
+    }
+
+    std::string get_measures_headers_as_str(std::string delimeter) {
+        std::string str = *measures_headers.begin();
+        for (auto it = measures_headers.begin() + 1; it != measures_headers.end(); it++) {
+            str += delimeter + *it;
+        }
+        return str;
+    }
+
+    void do_measures_n_print_w_msg(std::string message, std::string delimeter) {
```

Comment:
function name looks too complex and also miss any documentation. What n_print_w_msg means? What if name it "measure()"

Shouldn't delimiter be an unmodifiable constant? Scripts rely on it to be any whitespace character but if we allow modifying it one can choose ";" and scripts will stop working. Suggest remove this from API.

Reply:
Decided to split it on different functions. Delimiter has been hidden and defined within constant